project-1/gspread_code.py

```diff
@@ -0,0 +1,86 @@
+import os
+import logging 
+import gspread
+import pandas as pd
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+
+
+logging.basicConfig(level=logging.DEBUG, 
+                    format="%(levelname)s (%(asctime)s) : %(message)s (%(lineno)d)")
+ 
+current_working_directory = os.getcwd()
+#project_json is the credential file for the service account created
+project_json = "spartan-figure-402211-2241d0349aab.json"
+path = os.path.join(current_working_directory,project_json)
+
+#url for github repo stored in variable github_url
+github_url = "https://github.com/Humphery7?tab=repositories"
+# =============================================================================
+
+# loading credentials and authorizing gspread
+# client = gspread.service_account(filename = path)
+# # =============================================================================
```

Comment:
Do you still need this comment? If not, you can remove it

Reply:
yes there is something i don't understand, if i merge this branch with the main won't it edit the gspread_code.py, cause they are useful for the code, that is what connects to the site i am accessing, but for testing purposes i don't need them to be running and that is the reason i just commented them out

Reply:
I see, refactor your code with functions so that when you run your script or tests only the parts needed will run. Also, try to utilize :::: 

```
if __main__ == "__name__"
```

Reply:
i have tired both for now, i have not found a way yet, the issue is the driver variable has to be in the global namespace for me to be able to reference it in other functions in the class and also for me to be able to mock it easily with patch, 

cause once it is in the global namespace, it is regarded as the attribute of that module by patch, but putting it in a function will make it no longer that, 

and when i used if __name__ == __main__: the mocking function in the test code threw an error, i think cause it isnt seeing it again due to that, 
cause i am mocking the driver.find_element function in the test script, i will keep trying though, i just wanted to give an update on what i have tried so far